static/src/javascripts/projects/common/modules/identity/api.ts

```diff
@@ -413,14 +413,20 @@ export const updateUsername = (username: string): unknown => {
 };
 
 export const setConsent = (consents: SettableConsent): Promise<void> =>
-	fetch(`${idApiRoot}/users/me/consents`, {
-		method: 'PATCH',
-		credentials: 'include',
-		mode: 'cors',
-		body: JSON.stringify(consents),
-	}).then((resp) => {
-		if (resp.ok) return Promise.resolve();
-		return Promise.reject();
+	getAuthStatus().then((authStatus) => {
+		if (
+			authStatus.kind === 'SignedInWithCookies' ||
+			authStatus.kind === 'SignedInWithOkta'
+		) {
+			const authOptions = getOptionsHeadersWithOkta(authStatus);
+
+			return fetch(`${idApiRoot}/users/me/consents`, {
+				method: 'PATCH',
+				...authOptions,
+				mode: 'cors',
+				body: JSON.stringify(consents),
+			}).then((resp) => (resp.ok ? Promise.resolve() : Promise.reject()));
+		}
```

Comment:
I think the behaviour here has changed slightly.

Previously we would reject the promise if you were signed out, because `resp.ok` would be false.

I think what happens now is:
- if you are signed in, we make a request to the API
  - if `resp.ok == true`, resolve the promise
  - if `resp.ok == false`, reject the promise
- if you are not signed in, we don't make the fetch or return anything from the function. In this case we are resolving the promise to `undefined` implicitly. I think in the case where you are not signed in, we should reject the promise to indicate a failure, like:

```suggestion
	getAuthStatus().then((authStatus) => {
		if (
			authStatus.kind === 'SignedInWithCookies' ||
			authStatus.kind === 'SignedInWithOkta'
		) {
			const authOptions = getOptionsHeadersWithOkta(authStatus);

			return fetch(`${idApiRoot}/users/me/consents`, {
				method: 'PATCH',
				...authOptions,
				mode: 'cors',
				body: JSON.stringify(consents),
			}).then((resp) => (resp.ok ? Promise.resolve() : Promise.reject()));
		}
		return Promise.reject()
```

Another alternative is to "flatten the promise chain", like:

```suggestion
	getAuthStatus()
		.then((authStatus) =>
			authStatus.kind === 'SignedInWithCookies' ||
			authStatus.kind === 'SignedInWithOkta'
				? authStatus
				: Promise.reject('User is signed out'),
		)
		.then((authStatus) =>
			fetch(`${idApiRoot}/users/me/consents`, {
				method: 'PATCH',
				...getOptionsHeadersWithOkta(authStatus),
				mode: 'cors',
				body: JSON.stringify(consents),
			}).then((resp) => (resp.ok ? Promise.resolve() : Promise.reject())),
		);
```